packages/bvaughn-architecture-demo/components/Expandable.tsx

```diff
@@ -25,15 +25,23 @@ export default function Expandable({
   onChange?: (value: boolean) => void;
   useBlockLayoutWhenExpanded?: boolean;
 }) {
+  const [isPending, startTransition] = useTransition();
   const [isOpen, setIsOpen] = useState(defaultOpen);
 
   const onClick = (event: MouseEvent) => {
     event.stopPropagation();
 
+    if (isPending) {
+      return;
+    }
+
     const newIsOpen = !isOpen;
 
-    onChange(newIsOpen);
-    setIsOpen(newIsOpen);
+    // In case this change triggers a re-render that suspends, it should be in a transition.
+    startTransition(() => {
+      onChange(newIsOpen);
+      setIsOpen(newIsOpen);
+    });
```

Comment:
Also fixed a small suspense bug I uncovered while working on this change.

More context @ https://www.loom.com/share/0520050fad55416bbd9e394097e9aea7